launch/launch/substitutions/this_launch_file_dir.py

```diff
@@ -14,20 +14,31 @@
 
 """Module for the ThisLaunchFileDir substitution."""
 
+from typing import Iterable
 from typing import Text
 
 from .substitution_failure import SubstitutionFailure
+from ..frontend.expose import expose_substitution
 from ..launch_context import LaunchContext
+from ..some_substitutions_type import SomeSubstitutionsType
 from ..substitution import Substitution
 
 
+@expose_substitution('dirname')
 class ThisLaunchFileDir(Substitution):
     """Substitution that returns the absolute path to the current launch file."""
 
     def __init__(self) -> None:
         """Constructor."""
         super().__init__()
 
+    @classmethod
+    def parse(cls, data: Iterable[SomeSubstitutionsType]):
```

Comment:
@ivanpauno the actual problem is this method taking a `data` argument when it shouldn't

Reply:
All the methods have the same signature:
```python3
def parse(cls, data: Iterable[SomeSubstitutionsType]):
```
Regardless if data have 1, 2, 3, ..., 10 arguments.
Only deleting it for 0 arguments is strange. I preferred consistency.

Reply:
Fair enough. Not sure if `[ ]` is better than `None`.

Reply:
I prefer `[ ]`, so the parsing method only needs to check the len of the list, and not check for `None`.